src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutVertexNode.cs

```diff
@@ -0,0 +1,405 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.NativeFormat;
+using Internal.TypeSystem;
+using ILCompiler.DependencyAnalysisFramework;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Wrapper nodes for native layout vertex structures. These wrapper nodes are "abstract" as they do not 
+    /// generate any data. They are used to keep track of the dependency nodes required by a Vertex structure.
+    /// 
+    /// Any node in the graph that references data in the native layout blob needs to create one of these
+    /// NativeLayoutVertexNode nodes, and track it as a dependency of itself.
+    /// Example: MethodCodeNodes that are saved to the table in the ExactMethodInstantiationsNode reference 
+    /// signatures stored in the native layout blob, so a NativeLayoutPlacedSignatureVertexNode node is created
+    /// and returned as a static dependency of the associated MethodCodeNode (in the GetStaticDependencies API).
+    /// 
+    /// Each NativeLayoutVertexNode that gets marked in the graph will register itself with the NativeLayoutInfoNode,
+    /// so that the NativeLayoutInfoNode can write it later to the native layout blob during the call to its GetData API.
+    /// </summary>
+    internal abstract class NativeLayoutVertexNode : DependencyNodeCore<NodeFactory>
+    {
+        private Vertex _savedVertex;
+
+        public Vertex SavedVertex
+        {
+            get { return _savedVertex; }
+            protected set
+            {
+                Debug.Assert(_savedVertex == null || Object.ReferenceEquals(_savedVertex, value));
+                _savedVertex = value;
+            }
+        }
+
+        public override bool HasConditionalStaticDependencies => false;
+        public override bool HasDynamicDependencies => false;
+        public override bool InterestingForDynamicDependencyAnalysis => false;
+        public override bool StaticDependenciesAreComputed => true;
+        protected override string GetName() => null;
```

Comment:
Ah! I thought it had something to do with node names in the OBJ file, and I set it to null since we don't generate data for these nodes. I'll add a name to it